src/android/CodePush.java

```diff
@@ -270,6 +270,44 @@ private void cleanOldPackageSilently() {
         }
     }
 
+    private void clearDeploymentsIfBinaryUpdated() {
+        /* check if we have a deployed package already */
+        CodePushPackageMetadata deployedPackageMetadata = this.codePushPackageManager.getCurrentPackageMetadata();
+        if (deployedPackageMetadata != null) {
+            String deployedPackageTimeStamp = deployedPackageMetadata.nativeBuildTime;
+            long nativeBuildTime = Utilities.getApkEntryBuildTime(RESOURCES_BUNDLE, this.cordova.getActivity());
+            if (nativeBuildTime != -1) {
+                String currentAppTimeStamp = String.valueOf(nativeBuildTime);
+                if ((deployedPackageTimeStamp != null) && (currentAppTimeStamp != null)) {
```

Comment:
u can compress this and the following if with
`if ((deployedPackageTimeStamp != null) && !deployedPackageTimeStamp.equals(currentAppTimeStamp))`
afaik `.equals` returns null if the thing in the parentheses is null.